rhobs/alerting/data_plane/prometheus.gitops_sync_alerts.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: rhtap-gitops-sync-alerting
+  labels:
+    tenant: rhtap
+spec:
+  groups:
+  - name: gitops_sync_alerts
+    interval: 1m
+    rules:
+    - alert: GitOpsApplicationSyncErrors
+      expr: |
```

Comment:
The current expression is summarizing the apps across all env clusters (e.g. all staging host and member clusters). Is this the expected behavior?

Suppose that we have multiple apps out of sync in one stage member cluster. Do we want the alert to be fired when they are more than 5% of the apps within that cluster or between all stage clusters?

Reply:
See the use of `without` and `by` here: https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators

Reply:
@keithchong is summing across all clusters the expected behavior?

Reply:
@yftacherzog , I'm not sure if I can differentiate by cluster? Do you know what label to use?

Reply:
`source_cluster`.

Have a look at that metrics in https://grafana.app-sre.devshift.net/
It won't be your time series or clusters, but the label is the same.

Reply:
It looks like we might need a new rule for each cluster then, because I have to hardcode the name into the rule.

Reply:
You just need the label name rather than its value to include or exclude it from aggregation.

Reply:
Need to include the source_cluster in some test cases as well to see it's doing its part.

Reply:
I've added the test cases.  Note I had to update the alert rule as well otherwise, the first new test case that I added will fail.